### PR TITLE
Astro runtime 11.0

### PR DESCRIPTION
**Pull request by lzdanski**
- Astro runtime release (https://github.com/astronomer/astro-runtime/pull/1375)
- Airflow 2.9 Release notes (https://airflow.apache.org/docs/apache-airflow/2.9.0/release_notes.html#airflow-2-9-0-2024-04-08)
- Airflow 2.9 blog post (https://airflow.apache.org/blog/airflow-2.9.0/)

### CODE REVIEW

File: docs-partials/runtime-lifecycle-table.mdx [hunk 1]
```diff
@@ -6,6 +6,7 @@ The following table contains the exact lifecycle for each published version of A
 | [6](https://docs.astronomer.io/astro/runtime-release-notes#astro-runtime-600)  - Latest Patch (LTS) | 2.4             | September 19, 2022 | March 2024              |
 | [9](https://docs.astronomer.io/astro/runtime-release-notes#astro-runtime-900)  - Latest Patch (LTS) | 2.7             | August 18, 2023    | January 2025            |
 | [10](https://docs.astronomer.io/astro/runtime-release-notes#astro-runtime-1000) - Latest Patch      | 2.8             | December 18, 2023  | June 2024               |
+| [11](https://docs.astronomer.io/astro/runtime-release-notes#astro-runtime-1100) - Latest Patch      | 2.9             | April 8, 2024  | October 2024               |
```
Comment (lzdanski) [inline docs-partials/runtime-lifecycle-table.mdx:9]: Is the EOM date correct?

File: docs-partials/runtime-image-architecture.mdx [hunk 2]
```diff
@@ -112,6 +113,7 @@ astro dev bash pip freeze | grep astronomer-providers
 | 8             | 2.6                    | 3.10           |
 | 9             | 2.7                    | 3.11           |
 | 10            | 2.8                    | 3.11           |
```
Comment (lzdanski) [inline docs-partials/runtime-image-architecture.mdx:115]: Given the Pendulum caveat and airflow providers limitation, is this correct?

**Review by utkarsharma2 — APPROVED** [review]

File: docs-partials/runtime-lifecycle-table.mdx (hunk 1 above)
Comment (jedcunningham) [reply docs-partials/runtime-lifecycle-table.mdx:9]: Yes!

File: docs-partials/runtime-image-architecture.mdx [hunk 3]
```diff
@@ -131,6 +133,7 @@ The following table shows which versions Postgres are compatible with each versi
 | 8             | 2.6                    | 11-15             |
 | 9             | 2.7                    | 11-15             |
 | 10            | 2.8                    | 12-16             |
+| 11            | 2.9                    | 12-16             |
```
[the remark this answers is not in the document]
Comment (jedcunningham) [reply docs-partials/runtime-image-architecture.mdx:136]: Yep!

File: astro/runtime-release-notes.md [hunk 4]
```diff
@@ -29,6 +29,39 @@ If you're upgrading to receive a specific change, ensure the release note for th
...
+- Airflow version: 2.9.0
+
+### Airflow 2.9.0
+
+Astro Runtime 11.0.0 includes same-day support for Apache Airflow 2.9, which includes a number of new features and improvements. Airflow 2.9 includes the following changes:
+
+- New data-aware scheduling functionality includes using conditional logic (AND and OR) or combining both dataset dependencies and time-based schedules.
+- You can now create your own labels for dynamically mapped tasks with templates, instead of identifying them by searching through index numbers.
+- UI improvements that include filtering, viewing, and creating Datasets through the UI, the ability to localize `dag_id` and `task_id` names, task log grouping, and improvements to DAG-level views such as calendar, run duration, and the audit log.
+
```
Comment (phanikumv) [inline astro/runtime-release-notes.md:44]:
```suggestion
- AIP-58: Add object storage backend for xcom
```

File: docs-partials/runtime-image-architecture.mdx (hunk 2 above)
Comment (jedcunningham) [reply docs-partials/runtime-image-architecture.mdx:115]: No, the default will be 3.11 still.

File: docs-partials/runtime-image-architecture.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (utkarsharma2) [reply docs-partials/runtime-image-architecture.mdx:136]: Yup, it looks correct - https://pypi.org/project/apache-airflow/2.9.0/

File: astro/runtime-release-notes.md [hunk 5]
```diff
@@ -29,6 +29,39 @@ If you're upgrading to receive a specific change, ensure the release note for th
...
+### Upgrade to Python 3.12
+
+Airflow and the Astro Runtime now support Python 3.12. However, Pendulum 2 does not support Python 3.12. If you upgrade to Python 3.12 and want to use Airflow, you also need to upgrade to Pendulum 3.
+
+Refer to the [Airflow release notes](https://airflow.apache.org/docs/apache-airflow/2.9.0/release_notes.html#official-support-for-python-3-12-38025) for more information about any limitations in Python 3.12 support.
+
+### Additional improvements
+
+- New Listener API methods are considered stable and suitable for use in production. <!--https://github.com/apache/airflow/pull/36376-->
```
Comment (jedcunningham) [inline astro/runtime-release-notes.md:55]: Should these be actual links, or is this just for context for editors?

File: astro/runtime-release-notes.md [hunk 6]
```diff
@@ -29,6 +29,39 @@ If you're upgrading to receive a specific change, ensure the release note for th
...
+- Added the ability to automatically set a DAG to `off` after a pre-defined number of sequentially failed runs. <!--https://github.com/apache/airflow/pull/36935-->
```
Comment (jedcunningham) [inline astro/runtime-release-notes.md:56]: My 2c, "paused" is probably a better word to use than "off".

File: astro/runtime-release-notes.md
[the remark this answers is not in the document]
Comment (phanikumv) [reply astro/runtime-release-notes.md:1]: > We now support runtime slime image with 11.0.0. Lets document that

I do not think so ...I dont think  we need to publicize that...

cc @jedcunningham

File: docs-partials/runtime-image-architecture.mdx [hunk 7]
```diff
@@ -112,6 +113,7 @@ astro dev bash pip freeze | grep astronomer-providers
 | 8             | 2.6                    | 3.10           |
 | 9             | 2.7                    | 3.11           |
 | 10            | 2.8                    | 3.11           |
+| 11            | 2.9                    | 3.12           |
```
Comment (lzdanski) [inline docs-partials/runtime-image-architecture.mdx:116]:
```suggestion
| 11            | 2.9                    | 3.11           |
```

File: astro/runtime-release-notes.md (hunk 6 above)
Comment (lzdanski) [inline astro/runtime-release-notes.md:56]:
```suggestion
- Added the ability to automatically pause a DAG after a pre-defined number of sequentially failed runs.
```

File: astro/runtime-release-notes.md (hunk 5 above)
Comment (lzdanski) [reply astro/runtime-release-notes.md:55]: just context for editors! I'll remove them once we've got consensus

**Review by jwitz — CHANGES_REQUESTED** [review]
I have one comment about the additional improvements/ bug fixes sections, otherwise LGTM!

File: astro/runtime-release-notes.md [hunk 8]
```diff
@@ -29,6 +29,40 @@ If you're upgrading to receive a specific change, ensure the release note for th
...
+- Airflow version: 2.9.0
+
+### Airflow 2.9.0
+
+Astro Runtime 11.0.0 includes same-day support for Apache Airflow 2.9, which includes a number of new features and improvements. Airflow 2.9 includes the following changes:
+
+- New data-aware scheduling functionality includes using conditional logic (AND and OR) or combining both dataset dependencies and time-based schedules.
+- You can now create your own labels for dynamically mapped tasks with templates, instead of identifying them by searching through index numbers.
+- Added the ability to store XComs in an object storage supported backend.
+- UI improvements that include filtering, viewing, and creating Datasets through the UI, the ability to localize `dag_id` and `task_id` names, task log grouping, and improvements to DAG-level views such as calendar, run duration, and the audit log.
```
Comment (lzdanski) [inline astro/runtime-release-notes.md:44]:
```suggestion
- Delivered several significant improvements to the Airflow UI. For example, you can now filter, view, and create datasets through the Airflow UI..
```

File: astro/runtime-release-notes.md [hunk 9]
```diff
@@ -29,6 +29,37 @@ If you're upgrading to receive a specific change, ensure the release note for th
...
+- Delivered several significant improvements to the Airflow UI. For example, you can now filter, view, and create datasets through the Airflow UI.
+- New Listener API methods are considered stable and suitable for use in production.
+- Added the ability to automatically pause a DAG after a pre-defined number of sequentially failed runs.
+- Dataset URIs are validated when you enter them, and must conform to the rules set in AIP-60. See the [Dataset documentation](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/datasets.html) for more information.
+
+For more information about the major changes in this release, see the [Airflow Blog](https://airflow.apache.org/blog/airflow-2.9.0/) or the [Airflow release notes](https://airflow.apache.org/docs/apache-airflow/2.9.0/release_notes.html#airflow-2-9-0-2024-04-08).
+
+#### Upgrade to Python 3.12
+
+Astro Runtime now supports Python 3.12. However, [Pendulum](https://pendulum.eustace.io/) 2 does not support Python 3.12. If you upgrade to Python 3.12 and want to use Airflow, you also need to upgrade to Pendulum 3.
```
Comment (lzdanski) [inline astro/runtime-release-notes.md:53]:
```suggestion
Airflow now supports Python 3.12. However, [Pendulum](https://pendulum.eustace.io/) 2 does not support Python 3.12. If you upgrade to Python 3.12 and want to use Airflow, you also need to upgrade to Pendulum 3.
```

Comment (jwitz) [inline astro/runtime-release-notes.md]:
```suggestion
Airflow now supports Python 3.12. However, [Pendulum](https://pendulum.eustace.io/) 2 does not support Python 3.12. If you upgrade to Python 3.12 and want to use Airflow, you also need to upgrade to Pendulum 3.
```
Now that this is nested, we can use the original wording

File: astro/runtime-release-notes.md [hunk 10]
```diff
@@ -29,6 +29,37 @@ If you're upgrading to receive a specific change, ensure the release note for th
...
+
+For more information about the major changes in this release, see the [Airflow Blog](https://airflow.apache.org/blog/airflow-2.9.0/) or the [Airflow release notes](https://airflow.apache.org/docs/apache-airflow/2.9.0/release_notes.html#airflow-2-9-0-2024-04-08).
+
+#### Upgrade to Python 3.12
+
+Airflow now supports Python 3.12. However, [Pendulum](https://pendulum.eustace.io/) 2 does not support Python 3.12. If you upgrade to Python 3.12 and want to use Airflow, you also need to upgrade to Pendulum 3.
+
+Refer to the [Airflow release notes](https://airflow.apache.org/docs/apache-airflow/2.9.0/release_notes.html#official-support-for-python-3-12-38025) for more information about any limitations in Python 3.12 support.
+
+#### Airflow bug fixes
```
Comment (jwitz) [inline astro/runtime-release-notes.md:57]:
```suggestion
#### Bug fixes
```
Ok to cut since it's nested in the overall Airflow note

**Review by jwitz — APPROVED** [review]

File: astro/runtime-release-notes.md [hunk 11]
```diff
@@ -29,6 +29,37 @@ If you're upgrading to receive a specific change, ensure the release note for th
...
+
+- Fix a bug where after a task failed, and no longer exists in a DAG, you can now still access details about the DAG in the Grid View of the Airflow UI.([#38511](https://github.com/apache/airflow/pull/38511))
```
Comment (jwitz) [inline astro/runtime-release-notes.md:59]:
```suggestion
- Fix a bug where after a task failed, and no longer exists in a DAG, you can now still access details about the DAG in the Grid View of the Airflow UI.
```

File: astro/runtime-release-notes.md [hunk 12]
```diff
@@ -29,6 +29,37 @@ If you're upgrading to receive a specific change, ensure the release note for th
...
+- Fix a bug where Airflow would show `failed_upstream` when a dynamically mapped task was `skipped`.
```
Comment (jwitz) [inline astro/runtime-release-notes.md:60]:
```suggestion
- Fixed a bug where Airflow would show `failed_upstream` when a dynamically mapped task was `skipped`.
```

**Pull request merged**